### DISCUSSION TITLE
Can not find _dgemmt

### DISCUSSION THREAD

**Discussion by Xie-ry** [OpenMathLib/OpenBLAS#4410] [Q&A] [answered]
OpenBLAS provides the implementation of blas and lapack. 
SCALAPCK need blas.
MUMPS need SCALAPCK and LAPACK.

I don't want scalaback to use the built-in blas library when compiling. Is there a solution?

**Comment by martin-frbg** [comment]
Which version of Scalapack are you trying to compile ? As far as I know, you can tell it to use OpenBLAS at build time like `cmake -D USE_OPTIMIZED_LAPACK_BLAS=ON -D BLA_VENDOR=OpenBLAS` but see their own issue tracker at https://github.com/Reference-ScaLAPACK

**Reply by Xie-ry** [reply]
The problem has been resolved. I use the python script provided by scalapck to build.
like :  ./setup.py --blaslib=/lib64/libopenblas.so --lapacklib=/lib64/libopenblas.so

Now I get libscalapck.a.
But there's a new problem, can not find dgemmt_ symbol.

sgemmt_ or dgemmt_ seems seem to be provided by blas

**Comment by Xie-ry** [comment]
use "nm -D libopenblas.so", can see dgemm_.
what is the diif between dgemm_ and dgemmt_?

**Comment by martin-frbg** [comment, marked as answer]
GEMMT is a BLAS extension that performs the same operation as GEMM but only updates the upper or lower triangular part of the matrix. In OpenBLAS, this is only provided starting from version 0.3.23 - which release are you trying to use ?

**Reply by Xie-ry** [reply]
Thank you very much. The problem has been solved.
